### PR TITLE
OPSEXP-1862: add message broker support

### PR DESCRIPTION
**Pull request by alxgomz**
Ref: OPSEXP-1862

### CODE REVIEW

File: charts/alfresco-repository/templates/configmap.yaml [hunk 1]
```diff
@@ -2,12 +2,12 @@ apiVersion: v1
...
   name: >-
-    {{- $alfoptsCtx := dict "Values" (dict "nameOverride" "alfresco-opts") "Chart" .Chart "Release" .Release }}
-    {{ template "alfresco-repository.fullname" $alfoptsCtx }}
+    {{- $optsCtx := dict "Values" (dict "nameOverride" "alfresco-opts") "Chart" .Chart "Release" .Release }}
+    {{ template "alfresco-repository.fullname" $optsCtx }}
   labels:
     {{- include "alfresco-repository.labels" . | nindent 4 }}
 data:
-  ALFRESCO_OPTS: >-
+  CATALINA_OPTS: >-
```
Comment (gionn) [inline charts/alfresco-repository/templates/configmap.yaml:10]: remember to document the breaking change in acs-deployment

File: charts/alfresco-repository/values.yaml [hunk 2]
```diff
@@ -46,14 +46,37 @@ configuration:
...
-  # -- Alfresco java system properties. These properties must be provided as a
-  # string following the pattern "-Dproperty=value". They override the content
-  # of the global properties file but you should prefer using an existing
-  # configuration.repository.existingConfigMap.
-  ALFRESCO_OPTS: null
-  # -- Apache Tomcat command line options
+  # -- Java or Tomcat system properties. These properties must be provided as a
+  # single string following the pattern "-Dproperty=value -Dmoreprop=morevalue".
+  # They override the content of the global properties file but you should
+  # prefer using an existing configuration.repository.existingConfigMap.
```
Comment (gionn) [inline charts/alfresco-repository/values.yaml:79]:
```suggestion
  # prefer providing configuration.repository.existingConfigMap
```

File: charts/alfresco-repository/values.yaml [hunk 3]
```diff
@@ -46,14 +46,37 @@ configuration:
...
+    # -- Existing secret and their keys where to find the message broker
+    # username & password.
+    existingSecret:
+      # -- Name of a pre-existing secret containing database credentials
+      name: null
+      keys:
+        # -- Key within the secret holding the database username
+        username: BROKER_USERNAME
+        # -- Key within the secret holding the database password
+        password: BROKER_PASSWORD
```
Comment (gionn) [inline charts/alfresco-repository/values.yaml:65]:
```suggestion
      # -- Name of a pre-existing secret containing message broker credentials
      name: null
      keys:
        # -- Key within the secret holding the message broker username
        username: BROKER_USERNAME
        # -- Key within the secret holding the message broker password
        password: BROKER_PASSWORD
```

File: charts/alfresco-repository/values.yaml [hunk 4]
```diff
@@ -46,14 +46,37 @@ configuration:
...
     existingSecret: null
+  messageBroker:
+    # -- Message Broker URL
+    url: null
+    # -- Username to authenticate to the message broker
+    username: null
+    # -- Password to authenticate to the message broker
+    password: null
+    # -- Existing secret and their keys where to find the message broker
+    # username & password.
```
Comment (gionn) [inline charts/alfresco-repository/values.yaml:57]:
```suggestion
```

pretty much the same of what's documented below - and don't generate the ugly object entry in helm docs

File: charts/alfresco-repository/templates/secret-message-broker.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{{- if and (not .Values.configuration.messageBroker.existingSecret.name) .Values.configuration.messageBroker.username .Values.configuration.messageBroker.password }}
```
Comment (gionn) [inline charts/alfresco-repository/templates/secret-message-broker.yaml:1]: may not be of high impact, but the upstream activemq doesn't have any credentials enabled by default (or at least that what I remember), maybe we are too strict here? in the end generating a secret with empty credentials is still valid

**Review by gionn — CHANGES_REQUESTED** [review]

File: charts/alfresco-repository/templates/secret-message-broker.yaml (hunk 5 above)
Comment (alxgomz) [reply charts/alfresco-repository/templates/secret-message-broker.yaml:1]: Not generating the secret at all will result in the same outcome: empty variables si that will work with unauthenticated activemq.

File: charts/alfresco-repository/values.yaml (hunk 4 above)
Comment (alxgomz) [reply charts/alfresco-repository/values.yaml:57]: it slipped through

File: charts/alfresco-repository/templates/secret-message-broker.yaml (hunk 5 above)
Comment (gionn) [reply charts/alfresco-repository/templates/secret-message-broker.yaml:1]: true, and I see now the `optional: true` on the ref so it's cool

**Review by gionn — APPROVED** [review]

**Pull request merged**